### PR TITLE
Docs/add uploader docs

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**

### CODE REVIEW

File: docs/advanced-techniques.md [hunk 1]
```diff
@@ -88,3 +88,27 @@ A module is a standalone configuration object that can reference client code, re
 A merged module's code is treated the same as the application's code. The resulting application's bundles include the module code and configurations as if they were always a part of the application.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:88]: This is a strange paragraph. "is treated the same"? " as if they were always a part"?

File: docs/advanced-techniques.md [hunk 2]
```diff
@@ -88,3 +88,27 @@ A module is a standalone configuration object that can reference client code, re
...
 
 For an example on using modules, see the [Hacker News](https://github.com/reimagined/resolve/tree/master/examples/hacker-news) sample application. This application makes use of the authentication module and the comments module provided by reSolve.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:90]: "on using modules" Do you mean 'on how to use modules' or 'how modules are used'?

Comment (dirk-pieterse) [reply docs/advanced-techniques.md:90]: "makes use of the" So you mean it 'uses'?

Comment (dirk-pieterse) [reply docs/advanced-techniques.md:90]: "the comments module provided by reSolve" Does this need to be in passive voice?

File: docs/advanced-techniques.md [hunk 3]
```diff
@@ -88,3 +88,27 @@ A module is a standalone configuration object that can reference client code, re
...
+  defaultResolveConfig,
+  appConfig,
+  moduleAuth,
+  moduleUploader
+)
+```
+
+The **resolve-module-upload** module adds the following API endpoints to an application:
+
+- `/api/uploader/getFormUpload` - Returns an upload path to use in HTTP forms.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:110]: "an upload path to use in HTTP forms" Clarify

File: docs/advanced-techniques.md [hunk 4]
```diff
@@ -88,3 +88,27 @@ A module is a standalone configuration object that can reference client code, re
...
+- `/api/uploader/getUploadUrl` - Returns a path used to upload files.
+- `/api/uploader/getToken` - Given credentials returns the user's authorization token.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:112]: "Given credentials" Which 'given' credentials?

Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md:112]: Changed to "Takes user credentials and returns the user's authorization token."

File: docs/advanced-techniques.md (hunk 2 above)
Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md:90]: > "on using modules" Do you mean 'on how to use modules' or 'how modules are used'?

Changed to "on how to use modules"

Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md:90]: > "makes use of the" So you mean it 'uses'?

Yes, changed to "uses"

Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md:90]: Changed to "the comments module that reSolve provides"

File: docs/advanced-techniques.md (hunk 3 above)
Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md:110]: For example, if you add an "Upload File" form to your web application's page, you need to specify where to upload files when this form is submitted (upload path). The described endpoint returns this path.

File: docs/advanced-techniques.md (hunk 1 above)
Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md:88]: Rewrote the paragraph like so:
"A merged module's code is included into the resulting application's bundles and behaves as a part of the application."

File: docs/advanced-techniques.md [hunk 5]
```diff
@@ -85,6 +85,30 @@ A module is a standalone configuration object that can reference client code, re
 
 <!-- prettier-ignore-end -->
 
-A merged module's code is treated the same as the application's code. The resulting application's bundles include the module code and configurations as if they were always a part of the application.
+A merged module's code is included into the resulting application's bundles and behaves as a part of the application.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:88]: What do you mean "and behaves"?

File: docs/advanced-techniques.md [hunk 6]
```diff
@@ -85,6 +85,30 @@ A module is a standalone configuration object that can reference client code, re
...
 
-For an example on using modules, see the [Hacker News](https://github.com/reimagined/resolve/tree/master/examples/hacker-news) sample application. This application makes use of the authentication module and the comments module provided by reSolve.
+For an example on how to use modules, see the [Hacker News](https://github.com/reimagined/resolve/tree/master/examples/hacker-news) sample application. This application uses the authentication module and the comments module that reSolve provides.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:90]:
```suggestion
For an example on how to use modules, see the [Hacker News](https://github.com/reimagined/resolve/tree/master/examples/hacker-news) sample application. This application uses the authentication and comments modules from reSolve.
```

File: docs/advanced-techniques.md (hunk 3 above)
Comment (dirk-pieterse) [reply docs/advanced-techniques.md:110]:
```suggestion
- `/api/uploader/getFormUpload` - Returns the upload path used in HTTP forms.
```

File: docs/advanced-techniques.md (hunk 5 above)
Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md:88]: The code behaves

Comment (dirk-pieterse) [reply docs/advanced-techniques.md:88]: How can code 'behave'?

Comment (EugeniyBurmistrov) [reply docs/advanced-techniques.md]: Changed to "is interpreted as a part of the application"

File: docs/advanced-techniques.md [hunk 7]
```diff
@@ -85,6 +85,30 @@ A module is a standalone configuration object that can reference client code, re
 
 <!-- prettier-ignore-end -->
 
-A merged module's code is treated the same as the application's code. The resulting application's bundles include the module code and configurations as if they were always a part of the application.
+A merged module's code is included into the resulting application's bundles and is interpreted as a part of the application.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:88]:
```suggestion
A merged module's code is included in the resulting application's bundles.
```

File: docs/advanced-techniques.md [hunk 8]
```diff
@@ -85,6 +85,30 @@ A module is a standalone configuration object that can reference client code, re
...
+)
+```
+
+The **resolve-module-upload** module adds the following API endpoints to an application:
+
+- `/api/uploader/getFormUpload` - Returns an upload path to use in HTTP forms.
+- `/api/uploader/getUploadUrl` - Returns a path used to upload files.
+- `/api/uploader/getToken` - Takes user credentials and returns the user's authorization token.
+
+The [cli-uploader](https://github.com/reimagined/resolve/tree/master/examples/cli-uploader) example application demonstrates how to design a file uploader utility and handle file uploads on the server side.
```
Comment (dirk-pieterse) [inline docs/advanced-techniques.md:114]:
```suggestion
The [cli-uploader](https://github.com/reimagined/resolve/tree/master/examples/cli-uploader) example application demonstrates how to design a file uploader utility and handle file uploads on the server.
```

**Pull request merged**